tokio/tests/sync_broadcast.rs

```diff
@@ -640,3 +640,20 @@ fn send_in_waker_drop() {
     // Shouldn't deadlock.
     let _ = tx.send(());
 }
+
+#[test]
+fn broadcast_sender_closed() {
+    let (tx, rx) = broadcast::channel::<()>(1);
+    let rx2 = tx.subscribe();
+
+    let mut task = task::spawn(tx.closed());
+    assert_pending!(task.poll());
+
+    drop(rx);
+    assert!(!task.is_woken());
+    assert_pending!(task.poll());
+
+    drop(rx2);
+    assert!(task.is_woken());
+    assert_ready!(task.poll());
```

Comment:
Could you add another test that calls `subscribe` between dropping the last receiver and polling `closed()`, to see what happens in that case.

Reply:
Yeah, will do - sorry for the delay!

Reply:
Done - I simply re-open the channel if the `rx_cnt` is `0` when calling `new_receiver`. This allows us to re-create the `closed()` future, which I do in the new ` broadcast_sender_closed_with_extra_subscribe` test

Reply:
Hmm, it seems that this change made another test failing.

Now even after dropping all senders, the channel doesn't seem to be marked as closed when resubscribe is called afterwards.